core/src/test/java/org/apache/iceberg/TestManifestWriterVersions.java

```diff
@@ -213,27 +224,125 @@ public void testV2ManifestRewriteWithInheritance() throws IOException {
     checkRewrittenEntry(readManifest(manifest2), 0L, FileContent.DATA);
   }
 
+  @Test
+  public void testV3Write() throws IOException {
+    ManifestFile manifest = writeManifest(3);
+    checkEntry(
+        readManifest(manifest),
+        ManifestWriter.UNASSIGNED_SEQ,
+        ManifestWriter.UNASSIGNED_SEQ,
+        FileContent.DATA,
+        FIRST_ROW_ID);
+  }
+
+  @Test
+  public void testV3WriteWithInheritance() throws IOException {
```

Comment:
These tests are pretty hard for me to follow. I probably just haven't been in this code in a while but I can't tell what each one is trying to do

Reply:
Yeah, it's a bit weird because the write methods do more than one thing (write and also assert some things). I was trying to avoid refactoring the entire suite for this though.

Reply:
I'll spend some more time reading through them to see if I can follow then.